web/html/src/manager/shared/spa/spa-engine.js

```diff
@@ -22,7 +22,20 @@ window.pageRenderers.spa.init = function init() {
     }
   }]);
 
-  appInstance.on('endNavigate', function() {
+  appInstance.on('endNavigate', function(navigation) {
+    // workaround to redirect to the login page when there is no session:
+    // More info: https://github.com/liferay/senna.js/issues/302
+    const urlPath = appInstance.browserPathBeforeNavigate;
+    if(["Login.do", "/manager/login"].some(loginPath => urlPath && urlPath.includes(loginPath))) {
+      document.getElementsByClassName("spacewalk-main-column-layout")[0].innerHTML = `
```

Comment:
Mmmm, will this get any visibility? If we do a `window.location` reload right after this, I am afraid it will disappear, won't it?

Reply:
You will see it for 1/2 seconds. Basically, the redirect will be done twice due to a senna.js limitation on redirects: https://github.com/liferay/senna.js/issues/302

Reply:
I'm fine with the workaround, but either we keep the message visible for  some more time (let's say 5 seconds), or we should just hide it. Users would be scared to have missed to read a message if he can't read it fast enough. What do you think?